### PR TITLE
Optimization on division

### PR DESCRIPTION
**Pull request by Anurupa03**

### CODE REVIEW

File: core-contracts/FeeDistribution/src/main/java/finance/omm/score/fee/distribution/AbstractFeeDistribution.java [hunk 1]
```diff
@@ -60,13 +60,21 @@ private void distributeFeeToValidator(Address lendingPoolCoreAddr,BigInteger amo
...
                 "getActualUserDelegationPercentage", lendingPoolCoreAddr);
+        BigInteger amountToDistribute = amount;
+        BigInteger denominator = BigInteger.valueOf(100).multiply(ICX);
+        Address validatorAddr = lendingPoolCoreAddr;
         for (String validator : ommValidators.keySet()) {
-            Address validatorAddr = Address.fromString(validator);
-            BigInteger feePortion = (ommValidators.get(validator).multiply(amount)).divide(ICX).divide(BigInteger.valueOf(100));
+            validatorAddr = Address.fromString(validator);
+            BigInteger feePortion = (ommValidators.get(validator).multiply(amount)).divide(denominator);
+            amountToDistribute = amountToDistribute.subtract(feePortion);
```
Comment (nightowl121) [inline core-contracts/FeeDistribution/src/main/java/finance/omm/score/fee/distribution/AbstractFeeDistribution.java:69]: This implementation assumes to distribute the remaining amount to last validator. But it's better to use the progressive distribution  You would need to keep on deducting the total amount to distribute and the total(denominator).

File: core-contracts/FeeDistribution/src/main/java/finance/omm/score/fee/distribution/AbstractFeeDistribution.java [hunk 2]
```diff
@@ -60,13 +60,21 @@ private void distributeFeeToValidator(Address lendingPoolCoreAddr,BigInteger amo
...
             BigInteger feeAccumulatedAfterClaim = accumulatedFee.getOrDefault(validatorAddr, BigInteger.ZERO);
             accumulatedFee.set(validatorAddr, feeAccumulatedAfterClaim.add(feePortion));
 
         }
+        if (amountToDistribute.signum() > 0 && amountToDistribute.compareTo(amount) < 0 ){
+            BigInteger feeAccumulatedAfterClaim = accumulatedFee.getOrDefault(validatorAddr, BigInteger.ZERO);
```
Comment (nightowl121) [inline core-contracts/FeeDistribution/src/main/java/finance/omm/score/fee/distribution/AbstractFeeDistribution.java:75]: This approach also adds an additional read and write adding the cost.

File: core-contracts/FeeDistribution/src/main/java/finance/omm/score/fee/distribution/AbstractFeeDistribution.java (hunk 1 above)
Comment (Anurupa03) [reply core-contracts/FeeDistribution/src/main/java/finance/omm/score/fee/distribution/AbstractFeeDistribution.java]: Got it. Applied the changes and pushed. Please review it again

**Review by redlarva — APPROVED** [review]
looks good to me

**Review by nightowl121 — APPROVED** [review]
Approved

**Pull request merged**